core/services/relay/evm/types/codec_entry.go

```diff
@@ -146,11 +146,13 @@ func (entry *codecEntry) Init() (err error) {
 			}
 			for {
 				name = name + "_X"
+				args[i].Name = name
 				if !seenNames[name] {
 					break
 				}
 			}
 		}
+		args[i] = arg
```

Comment:
Doesn't this undo the earlier `args[i].Name = name` assignment in the for loop?

Reply:
I think you meant to set `entry.args[i] = arg` here, right?

Reply:
It's always the same slice in this case.  Unless there's a single unnamed arg, it's the same slice.

Reply:
Fixed the name.

Reply:
Oh, I think I see... so, unwrapNames only creates a new slice for `args` if it's a single unnamed arg, but the name only needs to be updated if there are multiple args right? Seems potentially brittle, but I guess that works.